gdal/frmts/netcdf/netcdfsgwriterutil.cpp

```diff
@@ -277,547 +408,554 @@ namespace nccfdriver
+         if(!itemsread)
+             return std::shared_ptr<OGR_SGFS_Transaction>(nullptr);
+
+         // If not, continue on and parse additional fields
+         switch(ntype)
+         {
+             // NC-3 Primitives
+             case NC_BYTE:
+                 return genericLogDataRead<OGR_SGFS_NC_Byte_Transaction, signed char>(varId, log);
+             case NC_SHORT:
+                 return genericLogDataRead<OGR_SGFS_NC_Short_Transaction, short>(varId, log);
+             case NC_INT:
+                 return genericLogDataRead<OGR_SGFS_NC_Int_Transaction, int>(varId, log);
+             case NC_FLOAT:
+                 return genericLogDataRead<OGR_SGFS_NC_Float_Transaction, float>(varId, log);
+             case NC_DOUBLE:
+                 return genericLogDataRead<OGR_SGFS_NC_Double_Transaction, double>(varId, log);
+#ifdef NETCDF_HAS_NC4
+             case NC_UBYTE:
+                 return genericLogDataRead<OGR_SGFS_NC_UByte_Transaction, unsigned char>(varId, log);
+             case NC_USHORT:
+                 return genericLogDataRead<OGR_SGFS_NC_UShort_Transaction, unsigned short>(varId, log);
+             case NC_UINT:
+                 return genericLogDataRead<OGR_SGFS_NC_UInt_Transaction, unsigned int>(varId, log);
+             case NC_INT64:
+                 return genericLogDataRead<OGR_SGFS_NC_Int64_Transaction, long long>(varId, log);
+             case NC_UINT64:
+                 return genericLogDataRead<OGR_SGFS_NC_UInt64_Transaction, unsigned long long>(varId, log);
+#endif
+             case NC_CHAR:
+             {
+                 size_t readcheck; // 0 means at least one read 0 bytes
+
+                 // Check what type of OP is requested
+                 int8_t op = 0;
+                 readcheck = VSIFReadL(&op, sizeof(int8_t), 1, log);
+                 if(!readcheck)
+                     return MTPtr(); // read failure
+
+                 size_t strsize;
+
+                 // get how much data to read
+                 readcheck = VSIFReadL(&strsize, sizeof(size_t), 1, log);
+                 if(!readcheck)
+                     return MTPtr(); // read failure
+
+                 std::unique_ptr<char, std::default_delete<char[]>> data(new char[strsize + 1]);
+                 memset(data.get(), 0, strsize + 1);
+
+                 // read that data and return it
+                 readcheck = VSIFReadL(data.get(), sizeof(char), strsize, log);
+                 if(!readcheck)
+                     return MTPtr(); // read failure
+
+                 // case: its a standard CHAR op
+                 if(!op)
+                 {
+                     return MTPtr(new OGR_SGFS_NC_Char_Transaction(varId, data.get()));  // data is copied so okay!
+                 }
+
+                 // case: its a CHARA op, additional processing
+                 else
+                 {
+                     size_t fullSizeCount;
+                     if(!VSIFReadL(&fullSizeCount, sizeof(size_t), 1, log))
```

Comment:
are there situations where fullSizeCount would be != strsize ?

Reply:
This was because of the whole truncation thing, but I guess it's been since disabled for CF-1.8 so no?

So I guess yeah we could change that.